infrastructure/testdriver/set_timezone.html

```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>TestDriver set_timezone method</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+
+<script>
+function defaultTimezone() {
+  return (new Intl.DateTimeFormat()).resolvedOptions().timeZone;
+}
+async_test(t => {
+  let timezone = "Asia/Taipei";
+  test_driver
+    .set_timezone(timezone)
+    .then(() => {
+      assert_equals(defaultTimezone(), timezone);
```

Comment:
Note: Technically if the current timezone of the machine running this test was *already* `Asia/Taipei`, then this test could pass for an invalid implementation of the feature (i.e. one that returned success but did nothing).

That said, I think that that is far too small an issue to worry much about, and so this seems fine to me as-is. (Unless there's a 'testing' timezone we could set it to that is guaranteed not to be the current timezone?)